configure

```
@@ -0,0 +1,169 @@
+OS_LIST="linux darwin"
+
+echo $OS_LIST | grep -w -i -q $OS
+if [ $? -ne 0 ]; then
+    echo "Error: unsupported OS $OS"
+    exit 1
+fi
+
+
+set_variable () {
+    echo "Setting $1 to $2"
+    grep "^$1" config/os-${OS}.make > /dev/null
+    if [ $? -ne 0 ]; then
+        echo "Error: dependency path variable $1 not found"
+        exit 1
+    fi
+    sed -i "s:\\(^$1[ ]*=\\).*:\1 $2:g" config/os-${OS}.make
+}
+
+set_openfst () {
+    echo "Setting up OpenFST at $1"
+    version=$(cat $1/configure.ac | head -n 1 | grep -oE "[0-9][0-9.]*[0-9]")
+    if [ $? -ne 0 ]; then
+        echo "Error: version information not found, please validate directory"
+        exit 1
+    fi
+    echo "found OpenFST version $version"
+    set_variable "OPENFSTDIR" $1
+    set_variable "OPENFST_VERSION" $version  
+}
+
+set_python () {
+    VERSION_STRING=$($1/python3 --version | grep -o "[0-9.]*")
+    if [ $? -ne 0 ];then exit 1;fi
+    VERSION_ARRAY=( ${VERSION_STRING//./ } )
+    MAJOR=${VERSION_ARRAY[0]}
+    MINOR=${VERSION_ARRAY[1]}
+    if [ $MAJOR -ne 3 ]; then
+        echo "Error: Expected Python version 3, found $MAJOR"
+        exit 1
+    fi
+    if [ $MINOR -ge 8 ]; then
+        # python version beginning from 3.8 do not include -lpython3.X when calling
+        # python3-config without --embed, so add it if such minor version is found
+        grep "python3-config --libs --embed" config/os-${OS}.make > /dev/null
+        if [ $? -ne 0 ]; then
+            echo "Python minor version is > 8, add --embed"
+            sed -i "s/\\(python3-config --libs\\)/\1 --embed/g" config/os-${OS}.make
+        else
+            echo "Python3: --embed already set"
+        fi
+    fi
+    PYTHON_BIN_PATH=$1
+    # make sure there is a trailing slash
+    [[ "${PYTHON_BIN_PATH}" != */ ]] && PYTHON_BIN_PATH="${PYTHON_BIN_PATH}/"
+    set_variable "PYTHON_BIN_PATH" $PYTHON_BIN_PATH
+}
+
+set_march () {
+    sed -i "s/-march=native/-march=$1/g" config/proc-x86_64.make
+}
```

if somebody wanted to use the i6 settings by default but change the architecture to XX, then `--i6` is executed first and sets this line to `-march=barcelona` and the subsequent call for `set_march XX` will fail silently because 
```
sed -i "s/-march=native/-march=$1/g" config/proc-x86_64.make
```
will not replace any lines.